api/rest/client/methods.go

```diff
@@ -172,3 +178,72 @@ func (c *Client) GetConnectGraph() (api.ConnectGraphSerial, error) {
 	err := c.do("GET", "/health/graph", nil, &graphS)
 	return graphS, err
 }
+
+// WaitFor is a utility function that allows for a caller to
+// wait for a paticular status for a CID. It returns a channel
+// upon which the caller can wait for the targetStatus.
+func (c *Client) WaitFor(ctx context.Context, ci *cid.Cid, local bool, target api.TrackerStatus, checkFreq time.Duration) (<-chan api.TrackerStatus, <-chan error) {
+	statusCh := make(chan api.TrackerStatus)
+	errCh := make(chan error)
+
+	go func() {
+		ticker := time.NewTicker(checkFreq)
+		defer ticker.Stop()
+
+	OUTER:
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				gblPinInfo, err := c.Status(ci, local)
```

Comment:
Use another function for this `reachedTarget(target, gblPinInfo) (bool, error)`.

Codeclimate will justly complain about too much complexity. Also, saves you using `goto` to outer loops.

Reply:
Also, you can put the glbPinInfo in the channel as you get it.

If target is reached, you close the channel are you're done.